apps/smart-contracts/core/test/DepositHook.test.ts

```diff
@@ -106,7 +106,9 @@ describe('=> DepositHook', () => {
       ).to.revertedWith('msg.sender != collateral')
     })
 
-    it('reverts if deposits not allowed', async () => {
+    it('reverts if user not allow list and deposits not allowed', async () => {
+      mockAllowlist.isIncluded.returns(false)
```

Comment:
```suggestion
      mockAllowlist.whenCalledWith(user.address).returns(false)
```
Lets utilize smocks ability to be more specific with when it reverts by passing in the specific user we are testing with.

Do this for the other testcases please.